extensions/amp-analytics/0.1/linker-manager.js

```diff
@@ -249,11 +268,61 @@ export class LinkerManager {
       const canonicalOrigin = this.urlService_.parse(canonicalUrl).hostname;
       if (!areFriendlyDomains(sourceOrigin, hostname)
           && !areFriendlyDomains(canonicalOrigin, hostname)) {
-        return;
+        return false;
       }
     }
+    return true;
+  }
+
+  /**
+   * Register callback that will handle form sumbits.
+   */
+  enableFormSupport_() {
+    this.formSubmitService_.then(formService =>
+      formService.beforeSubmit(this.handleFormSubmit_.bind(this)));
+  }
+
+  /**
+   * Check to see if any linker configs match this form's url, if so, send
+   * along the resolved linker value
+   * @param {HTMLFormElement} form
+   */
+  handleFormSubmit_(form) {
+    for (const linkerName in this.config_) {
+      const config = this.config_[linkerName];
+      const /** @type {Array} */ domains = config['destinationDomains'];
+
+      const url = form.getAttribute('action-xhr') ||
+          form.getAttribute('action');
+      const {hostname} = this.urlService_.parse(url);
+
+      if (this.isDomainMatch_(hostname, domains)) {
+        this.addDataToForm_(form, linkerName);
+      }
+    }
+  }
+
+
+  /**
+   * Add the linker pairs as <input> elements to form.
+   * @param {!Element} form
+   * @param {string} linkerName
+   */
+  addDataToForm_(form, linkerName) {
+    if (!this.resolvedLinkers_[linkerName]) {
+      return;
+    }
+
+    const attrs = dict({
+      'type': 'hidden',
+      'name': linkerName,
+      'value': this.resolvedLinkers_[linkerName],
+    });
 
-    el.href = addParamToUrl(href, name, this.resolvedLinkers_[name]);
+    const inputEl = createElementWithAttributes(
+        /** @type {!Document} */ (form.ownerDocument),
+        'input', attrs);
+    form.appendChild(inputEl);
```

Comment:
as discussed offline, we will need to treat method="GET" and method="POST" differently.

if method= "POST", we need to append the params to action URL, instead of hidden input.

Reply:
I will introduce in a follow up PR.